CS02-React&Redux.md

```diff
@@ -0,0 +1,127 @@
+
+- Redux was created by Dan Abramov and Andrew Clark in 2015. Abramov began writing the first Redux implementation while preparing for a conference talk at React Europe on hot reloading. Abramov remarks, "I was trying to make a proof of concept of Flux where I could change the logic. And it would let me time travel. And it would let me reapply the future actions on the code change."
+
+### 3. Who are the contributors to the project?
+
+- Facebook
+- Community
+    - React: 1240 Contributors
+    - ![](https://i.imgur.com/IpgTIFL.png)
+    - Redux: 613 Contributors
+    - ![](https://i.imgur.com/Ew7q7ed.png)
+
+
+### 4. Describe the project's funding. Is it funded? What kinds of support does it get?
+
+[Facebook Open Source](https://opensource.fb.com/)
+
+- Fundings
+- Labours
+- Tech Supports
+- Beer 🍺 & Coffee ☕️
+
+### 5. Describe the project's community. Are there separate developer / user communities? 
+
+- **React**
+    - [React Official Discuss](https://discuss.reactjs.org/) 	18.7k Posts
+    - [Stackoverflow](https://stackoverflow.com/questions/tagged/reactjs) 66.2k Watchers 100.9k Questions
+    - [Reddit](https://www.reddit.com/r/reactjs/) 69.4k Subscribers
+    - Others: 
+    [DEV’s React community](https://dev.to/t/react) | [Hashnode’s React community](https://hashnode.com/n/reactjs) | [Spectrum’s React community](https://spectrum.chat/react)
+
+- **Redux**
+    - [Stackoverflow](https://stackoverflow.com/questions/tagged/redux) 9.6k Watchers 15.4k Questions
+
+- **Reactiflux**
+    - [Homepage](https://www.reactiflux.com/) on Github
+    - [Discord](https://discord.gg/0ZcbPKXt5bZ6au5t) 37.7k Subscribers
+
+### 6. Where are the primary places they interact?
+
+- React:
+    - Facebook -> Github
+- Redux:
+    - Github
+
+### 7. Does the project have a Code of Conduct? What does it consist of? If it does not, why?
+
+- Yes
+- [Facebook Opensource CoD](https://code.fb.com/codeofconduct/)
+- Contents:
+    - Be friendly and patient.
+    - Be welcoming.
+    - Be considerat.
+    - Be respectful.
+    - Be careful in the words that you choose.
+    - We encourage everyone to participate and are committed to building a community for all. (Diversity Statement)
+    - Reporting Issues
+
+### 8. Does the project have a license? Why was the license chosen?
+
+- MIT (Be free and opened to everyone and every project whether it is commercial or not)
+- To be an universally popular frontend framework.
+- To be engaged by everyone.
+- *[Redux]* To convey the stateful concept to everyone. (Such as for Angular(NgRx), Vue(Vuex), etc.)
+
+### 9. Evaluate the project's approach to equity, diversity, and inclusion in its community. 
+
```

Comment:
Would you characterize the developer or user community of React as diverse? Are there any efforts or initiatives to welcome / include people from underrepresented groups in tech?

Reply:
Yes. There is one thing should be mentioned in the world **diversity** as we forgot to tell in class. The **diversity** I understand for *React* is that *React* actually brings web development to the whole stack of hybrid development. It highlights the concept of *ES6 Class* to embrace the combination of OOP programming, which is a breakthrough for it connects **Native App** concept with **Web**. This promotion is largely welcomed by both **Web Community** and **Native App Community** as it simplifies (or more accurately standardizes) the program standard in **Web**. And this is also a very important reason why it has larger community than most of other JS Library communities.